### PR TITLE
2023 NYR whole game day1

### LINKED ISSUE #124: 2023 NYR final touches to deck 1
- [ ] outline / "tentative plan"
- [ ] reference to cloud instance?
- [ ] install packages / package versions

### PR DESCRIPTION
**Pull request by hfrick**
This PR replaces the whole game analysis of the hockey data with a road map for day 1 in form of flowcharts only.

The "final" flowchart can serve as a reference for day 2: I would say day 2 dives deeper into the salmon colored part with better strategies for candidate model configurations via feature engineering and suggested tuning parameters - but the scaffolding around it remains.

If we like these, I'd add the corresponding stage at the end of the day-1 decks so that we can reiterate where in the road map we currently are.

Also: this isn't rendered yet, I wasn't sure if we're moving to the workflow of "render on main" (with main being `2023-nyr` here).

Edit: now also closes #124

### CODE REVIEW

**Comment by topepo** [comment]
FYI... I was rendering the main nyr branch and had errors due to old files that had been deleted (roc curve files for the hockey models). I edited this file on that branch, which caused a merge conflict here. I've fixed it.

File: slides/01-introduction.qmd [hunk 1]
```diff
@@ -141,3 +141,116 @@ inference vs. prediction
 library(tidymodels)
 ```
 
+##  {background-image="images/tm-org.png" background-size="contain"}
+
+## The whole game
```
Comment (topepo) [inline slides/01-introduction.qmd:146]: Can you have a slide before this that describes what we are about to show? Something like "we want you to think about 'the whole game' of modeling. That means... "

File: slides/01-introduction.qmd [hunk 2]
```diff
@@ -141,3 +141,116 @@ inference vs. prediction
...
+
+```{r diagram-split, echo = FALSE}
+#| fig-align: "center"
+
+knitr::include_graphics("images/whole-game-split.jpg")
+```
+
+## The whole game
```
Comment (topepo) [inline slides/01-introduction.qmd:154]: We color coordinate based on data in a lot of places. The colors of the resamples and models are more similar to the test set color than the training set. 

Can you change that? Maybe use the colors that we have in the tidymodels book and in the later notes? 

I think that consistency is important here even for something subtle like the shapes and colors.

File: slides/01-introduction.qmd [hunk 3]
```diff
@@ -141,3 +141,116 @@ inference vs. prediction
...
+
+```{r diagram-model-n, echo = FALSE}
+#| fig-align: "center"
+
+knitr::include_graphics("images/whole-game-model-n.jpg")
+```
+
+## The whole game
```
Comment (topepo) [inline slides/01-introduction.qmd:170]: Maybe add some notes or make a change to the diagram that reflects the iterative nature of resampling. 

Looking at this, I think that people will think that you are just fitting the model to the training set and moving on.

File: slides/01-introduction.qmd [hunk 4]
```diff
@@ -141,3 +141,116 @@ inference vs. prediction
...
+```
+
+
+## Let's install some packages
+
+If you are using your own laptop instead of RStudio Cloud:
+
+```{r}
+#| eval: false
+install.packages(c("Cubist", "DALEXtra", "doParallel", "earth", "embed",
```
Comment (topepo) [inline slides/01-introduction.qmd:209]: The main page for the `2023-nyr` branch has

```r
install.packages("pak")

# Then the packages for both days
pkgs <- 
  c("bonasi", "doParallel", "finetune", "lightgbm", "lme4", "plumber", 
    "probably", "ranger", "rpart", "rpart.plot", "stacks", "textrecipes", 
    "tidymodels", "tidymodels/modeldatatoo", "vetiver")
pak::pak(pkgs)
```

File: slides/01-introduction.qmd [hunk 5]
```diff
@@ -141,3 +141,116 @@ inference vs. prediction
...
+remotes::install_github("topepo/ongoal@hockeyR")
+```
+
+. . .
+
+<br></br>
+
+### <i class="fa fa-cloud"></i> Or log in to RStudio Cloud:
+
+[bit.ly/tidymodels-iceland-2022](http://bit.ly/tidymodels-iceland-2022)
```
Comment (topepo) [inline slides/01-introduction.qmd:223]: I will make a cloud project tomorrow and will update this link.

File: slides/01-introduction.qmd [hunk 6]
```diff
@@ -141,3 +141,116 @@ inference vs. prediction
...
+
+
+## Our versions
+
+```{r pkg-list, echo = FALSE}
+deps <- c("DALEXtra", "doParallel", "embed", "forcats", "ranger", "remotes", "rpart",
```
Comment (topepo) [inline slides/01-introduction.qmd:229]: Update these based on the changes above.

**Review by topepo — CHANGES_REQUESTED** [review]

File: slides/01-introduction.qmd (hunk 2 above)
Comment (hfrick) [reply slides/01-introduction.qmd:154]: I can definitely adapt the colors 👍 If aiming for the colors of the book/later notes: do you have the hex color code for those or would that mean dipping into OmniGraffle?

File: slides/01-introduction.qmd (hunk 3 above)
Comment (hfrick) [reply slides/01-introduction.qmd:170]: Are you refering to the flow chart in [whole-game-model-n.jpg](https://github.com/tidymodels/workshops/pull/137/files#diff-9feda7e8a16db53708b63fa04797d3766c4eda250f24d4e234d97f976fc286f2)? Or the final version of the flow chart?

For the `whole-game-model-n` version:
My motivation for showing one fit on the entire training set (the chart one prior to this) was that this is what we are doing in the "What makes a model" deck and I'd like to end decks 2-4 with the corresponding stage of this diagram series for the whole game. The intension of this one is to motivate resamples. If it's too tempting to read this as "we are fitting several models straight to the training data", we could skip `model-n` and go straight to `whole-game-resamples`.

For the final version:
Do you mean with iterativeness that the resampling scheme typically involves multiple different splits of the training data? I.e. the "sliced" visualization of the training data seen in some omnigraffle version from previous years?

Comment (topepo) [reply slides/01-introduction.qmd:170]: > If it's too tempting to read this as "we are fitting several models straight to the training data"

I think that having some `:::notes` for the speakers to tell the audience that were are not doing a single fit on the entire training set would be enough.

File: slides/01-introduction.qmd (hunk 2 above)
Comment (topepo) [reply slides/01-introduction.qmd:154]: The yellowish color for the training set is #ffe9bb and the blueish color for the test set is #e5e6ff

File: slides/01-introduction.qmd (hunk 4 above)
Comment (simonpcouch) [reply slides/01-introduction.qmd:209]: Related: changes here should reflect https://github.com/tidymodels/workshops/pull/139!

**Review by simonpcouch — APPROVED** [review]
I really dig these diagrams!

**Review by EmilHvitfeldt — APPROVED** [review]

File: slides/01-introduction.qmd (hunk 2 above)
Comment (hfrick) [reply slides/01-introduction.qmd:171]: Thanks! New colors in [73dd7b4](https://github.com/tidymodels/workshops/pull/137/commits/73dd7b44773143ccb0611c7af4d1239fb0fa0e75)

File: slides/01-introduction.qmd (hunk 5 above)
Comment (hfrick) [reply slides/01-introduction.qmd]: I've changed the link in the slides to `http://bit.ly/tidymodels-nyr-2023`. If you make the bitly link point to the new clour project, we should be set!

**Comment by hfrick** [comment]
I'm gonna transfer the changes to a fresh PR, and leaving the previous whole-game plots/charts untouched
